include/alpaka/mem/buf/cpu/Set.hpp

```diff
@@ -151,9 +151,17 @@ namespace alpaka
                             {
                                 meta::ndLoopIncIdx(
                                     extentWithoutInnermost,
+
+                                    // workaround for HIP(HCC) to
+                                    // avoid forbidden host-call
+                                    // within host-device functions
+                                    #if defined(BOOST_COMP_HCC) && BOOST_COMP_HCC
+                                    ALPAKA_FN_HOST_ACC
```

Comment:
Can we add this attribute unconditionally?

Reply:
not directly, because CUDA cannot handle this annotation in front of a lambda function, while HCC requires it due to the more rigid host-device restrictions. There is a chain of functions, where at least one of them is annotated with `ALPAKA_FN_HOST_ACC`, and this is passed as a functor, thus required to be `ALPAKA_FN_HOST_ACC`.